contracts/terraswap_router/src/contract.rs

```diff
@@ -225,48 +223,23 @@ pub fn query_config(deps: Deps) -> StdResult<ConfigResponse> {
 }
 
 fn simulate_swap_operations(
-    deps: Deps,
+    deps: Deps<Empty>,
     offer_amount: Uint128,
     operations: Vec<SwapOperation>,
 ) -> StdResult<SimulateSwapOperationsResponse> {
     let config: Config = CONFIG.load(deps.storage)?;
     let terraswap_factory = deps.api.addr_humanize(&config.terraswap_factory)?;
-    let terra_querier = TerraQuerier::new(&deps.querier);
 
     let operations_len = operations.len();
     if operations_len == 0 {
         return Err(StdError::generic_err("must provide operations"));
     }
 
-    let mut operation_index = 0;
     let mut offer_amount = offer_amount;
     for operation in operations.into_iter() {
-        operation_index += 1;
-
         match operation {
-            SwapOperation::NativeSwap {
-                offer_denom,
-                ask_denom,
-            } => {
-                // Deduct tax before query simulation
-                // because last swap is swap_send
-                if operation_index == operations_len {
-                    offer_amount = offer_amount.checked_sub(compute_tax(
-                        &deps.querier,
-                        offer_amount,
-                        offer_denom.clone(),
-                    )?)?;
-                }
-
-                let res: SwapResponse = terra_querier.query_swap(
-                    Coin {
-                        denom: offer_denom,
-                        amount: offer_amount,
-                    },
-                    ask_denom,
-                )?;
-
-                offer_amount = res.receive.amount;
+            SwapOperation::NativeSwap { .. } => {
+                return Err(StdError::generic_err("does not support native_swap"));
```

Comment:
can we update README.md following this change?

Reply:
how about just removing `NativeSwap`? seems tax querier already has been removed

Reply:
https://github.com/terraswap/terraswap/pull/44/commits/cfb89576757bb1a0f1f5a4b76061facd1a3380b3

Reply:
https://github.com/terraswap/terraswap/pull/44/commits/7e5fd134ec0a38d0af549bba7c36fc47ad6bd38d

Reply:
needs to update schema as well 🙇‍♀️

Reply:
https://github.com/terraswap/terraswap/pull/44/commits/6c729775cf67596ab9b50b47a29a1be2362b1d23, https://github.com/terraswap/terraswap/pull/44/commits/43644e09cc40244ee21eb580f43f3d951abcbded